Toolchain/BuildJakt.sh

```diff
@@ -0,0 +1,193 @@
+    echo SYSROOT is "$SYSROOT"
+
+    mkdir -p "$DIR/Tarballs"
+
+    JAKT_COMMIT_HASH="f7ab7725114538f0668c58973bab2cf52b8e121c"
+    JAKT_NAME="jakt-${JAKT_COMMIT_HASH}"
+    JAKT_TARBALL="${JAKT_NAME}.tar.gz"
+    JAKT_GIT_URL="https://github.com/serenityos/jakt"
+
+    # === DEPENDENCIES ===
+    buildstep dependencies echo "Checking whether 'ninja' is available..."
+    if ! command -v "${NINJA:-ninja}" >/dev/null; then
+        buildstep dependencies echo "Please make sure to install Ninja (for the '${NINJA:-ninja}' tool)."
+        exit 1
+    fi
+
+    buildstep dependencies echo "Checking whether 'cmake' is available..."
+    if ! command -v cmake >/dev/null; then
+        buildstep dependencies echo "Please make sure to install CMake (for the 'cmake' tool)."
+        exit 1
+    fi
+
+    buildstep dependencies echo "Checking whether your C++ compiler works..."
+    if ! ${CXX:-c++} -o /dev/null -xc++ - >/dev/null <<'PROGRAM'
+int main() {}
+PROGRAM
+    then
+        buildstep dependencies echo "Please make sure to install a working C++ compiler."
+        exit 1
+    fi
+
+    # === DOWNLOAD AND PATCH ===
+    pushd "$DIR/Tarballs"
+        if [ ! -d "${JAKT_NAME}" ]; then
+            echo "Downloading jakt..."
+            rm -f "$JAKT_TARBALL"
+            curl -L --output "$JAKT_TARBALL" "$JAKT_GIT_URL/archive/$JAKT_COMMIT_HASH.tar.gz"
+
+            echo "Extracting jakt..."
+            tar -xzf "${JAKT_TARBALL}"
+        else
+            echo "Using existing Jakt source directory"
+        fi
+    popd
+
+
+    # === COMPILE AND INSTALL ===
+
+    rm -rf "$PREFIX"
+    mkdir -p "$PREFIX"
+
+    mkdir -p "$DIR/Build/$ARCH"
+
+    pushd "$DIR/Build/$ARCH"
+        echo "XXX serenity libc headers"
+        mkdir -p "$BUILD"
+        pushd "$BUILD"
+            mkdir -p Root/usr/include/
+            SRC_ROOT=$($REALPATH "$DIR"/..)
+            FILES=$(find \
+                "$SRC_ROOT"/AK \
+                "$SRC_ROOT"/Kernel/API \
+                "$SRC_ROOT"/Kernel/Arch \
+                "$SRC_ROOT"/Userland/Libraries/LibC \
+                "$SRC_ROOT"/Userland/Libraries/LibELF/ELFABI.h \
+                "$SRC_ROOT"/Userland/Libraries/LibRegex/RegexDefs.h \
+                -name '*.h' -print)
+            for header in $FILES; do
+                target=$(echo "$header" | sed \
+                    -e "s|$SRC_ROOT/AK/|AK/|" \
+                    -e "s|$SRC_ROOT/Userland/Libraries/LibC||" \
+                    -e "s|$SRC_ROOT/Kernel/|Kernel/|" \
+                    -e "s|$SRC_ROOT/Userland/Libraries/LibELF/|LibELF/|" \
+                    -e "s|$SRC_ROOT/Userland/Libraries/LibRegex/|LibRegex/|")
+                buildstep "system_headers" mkdir -p "$(dirname "Root/usr/include/$target")"
+                buildstep "system_headers" "$INSTALL" "$header" "Root/usr/include/$target"
+            done
+            unset SRC_ROOT
+        popd
+
+        rm -rf jakt
+        mkdir -p jakt
+
+        pushd jakt
```

Comment:
surely we only need to build jakt that knows how to target $ARCH once, and then build_for should only apply to the shupport libs?  Or do we encode the sysroot in the jakt binary?

Reply:
Or... are there compiler-specific dependencies in actually executing the jakt binary itself? surely not. only the support libs should be compiler-specific.... right?

Reply:
(yes, only the support libs are compiler-dependent)